python/ray/runtime_env.py

```diff
@@ -24,8 +24,8 @@
 def _build_proto_pip_runtime_env(runtime_env_dict: dict, runtime_env: ProtoRuntimeEnv):
     """Construct pip runtime env protobuf from runtime env dict."""
     if runtime_env_dict.get("pip"):
-        if isinstance(runtime_env_dict["pip"], list):
-            runtime_env.python_runtime_env.pip_runtime_env.config.packages.extend(
+        if isinstance(runtime_env_dict["pip"], dict):
+            runtime_env.python_runtime_env.pip_runtime_env.config = json.dumps(
```

Comment:
Can we avoid serialization in python? We can add two fields to `PipRuntimeEnv` protobuf.

Reply:
done.